tests/console/velociraptor_client_event_collect.pm

```diff
@@ -47,79 +47,175 @@ sub run {
+    script_run 'echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEArandomstring test@testkey" >> ~/.ssh/authorized_keys && chmod 700 ~/.ssh && chmod 600 ~/.ssh/authorized_keys';
+    # trigger SUSE.Linux.Events.TCPConnections
+    script_run "nc localhost 8080";
+    # trigger SUSE.Linux.Events.Timers
+    # trigger SUSE.Linux.Events.UserAccount
+    # trigger SUSE.Linux.Events.UserGroupMembershipUpdates
 
     # get client id
-    sleep 10;
     my $clientid = script_output('velociraptor-client --api_config ~/api.config.yaml query \'SELECT *, os_info.hostname as Hostname, client_id FROM clients()\' | grep -oP \'"client_id": "\K.*(?=")\'', 120);
 
-    # add client monitoring
-    my @artifacts = qw(SUSE.Linux.Events.DNS SUSE.Linux.Events.ExecutableFiles SUSE.Linux.Events.ImmutableFile SUSE.Linux.Events.NewFiles SUSE.Linux.Events.NewFilesNoOwner SUSE.Linux.Events.NewHiddenFile SUSE.Linux.Events.NewZeroSizeLogFile SUSE.Linux.Events.Packages SUSE.Linux.Events.ProcessStatuses SUSE.Linux.Events.SSHLogin SUSE.Linux.Events.Services SUSE.Linux.Events.SshAuthorizedKeys SUSE.Linux.Events.SystemLogins SUSE.Linux.Events.TCPConnections SUSE.Linux.Events.Timers SUSE.Linux.Events.UserAccount SUSE.Linux.Events.UserGroupMembershipUpdates);
+    # check for collected event on server
+    sleep 90;
     foreach my $artifact (@artifacts) {
-        assert_script_run("velociraptor-client --api_config ~/api.config.yaml query 'SELECT add_client_monitoring(artifact=\"$artifact\") FROM scope()'");
+        my $path = "/var/tmp/velociraptor/clients/$clientid/monitoring/$artifact/";
+        if (-d $path) {
+            print "$path event logs present";
+            script_run("ls $path | grep json");
+        }
+        else {
+            print "$path event logs missing";
+        }
     }
 
-    # generetae client events
-    sleep 10;
-    my $i = 0;
-    while ($i < 10) {
-        assert_script_run "echo 'Client Event exec' >> /home/genfile$i.sh";
-        assert_script_run "chmod +x /home/genfile$i.sh";
-        assert_script_run "echo 'Client Event exec' >> /tmp/genfile$i.sh";
-        assert_script_run "chmod +x /tmp/genfile$i.sh";
-        $i++;
-    }
-    sleep 60;
-    # check for collected event on server
-    script_output "ls -la /var/tmp/velociraptor/clients/$clientid/monitoring_logs/";
-    script_output "ls -la /var/tmp/velociraptor/clients/$clientid/monitoring/";
-    my $files = script_output "find /var/tmp/velociraptor/clients/$clientid/monitoring/ -type f -name '*.json' | awk -F/ '{print \$F}'";
-    my @names = split /\s+/, $files;
-    for (@names) {
-        my @name = split /\//, $_;
-        upload_logs($_, log_name => "artifact-$name[7].json");
+    # upload event logs
+    script_run "tar cvpzf eventlogs.tgz /var/tmp/velociraptor/clients/$clientid/";
+    upload_logs('eventlogs.tgz');
+
+    # search for SUSE.Linux.Events.ExecutableFiles
+    script_output "grep -r -i genfile* /var/tmp/velociraptor/clients/$clientid/*";
+    if (!is_sle('=12-SP5')) {
```

Comment:
suggestion: group the skipping of eBPF plugin checks and comment why. Also we are missing a check for DNS - maybe `dig` something other than localhost.
```
# The eBPF based plugins don't get built on SLE12SP5
if (!is_sle('=12-SP5')) {
        # search for SUSE.Linux.Events.ImmutableFile
        script_output "grep -r -i immutible* /var/tmp/velociraptor/clients/$clientid/*";
        # search for SUSE.Linux.Events.TCPConnections
        script_output "grep -r -i localhost* /var/tmp/velociraptor/clients/$clientid/*";
        # search for SUSE.Linux.Events.DNS
        # TODO
}
```